broker/src/tunneldigger_broker/broker.py

```diff
@@ -77,6 +83,27 @@ def create_tunnel(self, broker, address, uuid, remote_tunnel_id, client_features
         else:
             tunnel_str = "(%s)" % uuid
 
+        # Rate limit creation of new tunnels with the same IP address.
+        # Runs broker.tunnel-connection-threshold hook if threshold exceeded:
+        if self.connection_rate_limit_per_ip_count > 0 and self.connection_rate_limit_per_ip_time > 0:
+            if address[0] not in self.last_tunnel_created_per_ip:
+                self.last_tunnel_created_per_ip[address[0]] = deque([], self.connection_rate_limit_per_ip_count) # Create deque collection if IP address does not exist in dict
```

Comment:
That's cool, I did not know `deque` had a "maxlen" parameter. :D
Do you even still need the `popleft` below?